packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useControlledValue, useIdAllocator } from '@leafygreen-ui/hooks';
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import { Label } from '@leafygreen-ui/typography';
+
+import { InputIcon } from '../InputIcon';
+import { StateNotifications } from '../StateNotifications';
+import { TogglePassword } from '../TogglePassword';
+
+import {
+  inputBaseStyles,
+  inputBaseThemeStyles,
+  inputDisabledBaseStyles,
+  inputDisabledThemeStyles,
+  inputIconSizeStyles,
+  inputSizeStyles,
+  inputThemeStyles,
+  inputWrapperStyles,
+  labelBaseStyles,
+  labelLargeOverrideStyles,
+} from './PasswordInput.styles';
+import {
+  type NotificationProps,
+  PasswordInputProps,
+  SizeVariant,
+  States,
+} from './PasswordInput.types';
+import { getStateFromArray } from './utils';
+
+export const PasswordInput = React.forwardRef<
+  HTMLInputElement,
+  PasswordInputProps
+>(
+  (
+    {
+      value: valueProp,
+      onChange: onChangeProp,
+      className,
+      darkMode: darkModeProp,
+      label,
+      sizeVariant = SizeVariant.Default,
+      stateNotifications = [],
+      disabled = false,
+      autoComplete = 'new-password',
+      id: idProp,
+      'aria-describedby': ariaDescribedbyProp,
+      'aria-labelledby': ariaLabelledbyProp,
+      ...rest
+    }: PasswordInputProps,
+    forwardedRef,
+  ) => {
+    const [showPassword, setShowPassword] = useState<boolean>(false);
+    const prefix = 'passwordinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const ariaDescribedby = useIdAllocator({
+      prefix,
+      id: ariaDescribedbyProp,
+    });
+    const ariaLabelledby = useIdAllocator({
+      prefix,
+      id: ariaLabelledbyProp,
+    });
+    const { theme, darkMode } = useDarkMode(darkModeProp);
+    const { value, handleChange } = useControlledValue(valueProp, onChangeProp);
+
+    if (!label && !ariaLabelledbyProp && !rest['aria-label']) {
+      console.warn(
+        'For screen-reader accessibility, label, aria-labelledby, or aria-label must be provided',
+      );
+    }
+
+    // If disabled then hide password
+    useEffect(() => {
+      if (disabled) setShowPassword(false);
+    }, [disabled]);
+
+    const handleTogglePasswordClick = () => setShowPassword(s => !s);
+
+    /**
+     * The overall state of the component
+     */
+    const state: States = Array.isArray(stateNotifications)
```

Comment:
when would stateNotifications not be an array?

Reply:
if `aria-describedby` is passed, instead of an array of message, a string with the input state is expected.

Reply:
but you're setting a default for stateNotifications to be an empty array -- so it will always be an array

Reply:
i'm a little confused, stateNotifications can either be an array or a string(one of the `States`). If a string is passed to stateNotifications then that would override the default, right?

Reply:
I think Brooke is asking, when would someone explicitly pass in just a string vs an array? 
I think I understand, but the api is a bit confusing at first glance (passing in a single state string to a prop called `stateNotifications` feels counter intuitive)

e.g. If I want a `valid` password input with an external message, I would have to write:
```tsx
<PasswordInput
  aria-describedby="my-description"
  stateNotifications="valid"
/>
<span id="my-description" />
```

Reply:
You could use some TS trickery where if `aria-describedby` is set, then we expect a prop called `state`